scripts/auto_archive.py

```diff
@@ -0,0 +1,224 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import os
+from datetime import datetime
+from datetime import timedelta
+
+import git
+import humanfriendly
+from datalad.plugin import export_archive
+from github import Github
+
+from scripts.datalad_utils import get_dataset
+from scripts.datalad_utils import install_dataset
+from scripts.log import get_logger
+from tests.functions import get_proper_submodules
+
+
+logger = get_logger(
+    "CONP-Archive", filename="conp-archive.log", file_level=logging.DEBUG
+)
+
+
+class ArchiveFailed(Exception):
+    pass
+
+
+def parse_args():
+    example_text = """Example:
+    PYTHONPATH=$PWD python scripts/auto_archive.py <out_dir>
+    """
+
+    parser = argparse.ArgumentParser(
+        description="Archiver for the CONP-datasets.",
+        epilog=example_text,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "out_dir", type=str, help="Path to store the archived datasets."
+    )
+    parser.add_argument(
+        "--max-size",
+        type=float,
+        default=20.0,
+        help="Maximum size of dataset to archive in GB.",
+    )
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--all",
+        action="store_true",
+        help="Archive all the datasets rather than those modified since the last time.",
+    )
+    group.add_argument(
+        "--dataset",
+        "-d",
+        type=str,
+        nargs="+",
+        help="Restrict the archive to the datasets specified.",
```

Comment:
should probably specify here that `projects/` need to be in front of the dataset name or that the dataset specified should be the relative path to the dataset?

Reply:
@cmadjar, I modified the help for the argument to specify that it expect datasets *paths*. I also change the script to allow for either absolute or relative paths.